kyaml/openapi/openapi.go

```diff
@@ -547,6 +547,9 @@ func SetSchema(openAPIField map[string]string, schema []byte, reset bool) error
 		return nil
 	}
 
+	// if the schema is changed, initSchema should parse the new schema
+	defer func() { globalSchema.schemaInit = false }()
```

Comment:
Why do we need to do this in a deferred func instead of setting it inline? Do we still want the reset in cases where SetSchema returned an error?

Reply:
done, inlined it in the necessary places rather than deferring it